egs2/TEMPLATE/cls1/README.md

```diff
@@ -0,0 +1,271 @@
+```
+This might help you understand each stage's processing and directory structure.
+
+### Evaluation
+
+Here we show the example command to calculate classification metrics:
+
+```sh
+
+cd egs2/<recipe_name>/cls1
+. ./path.sh
+
+python3 pyscripts/utils/cls_score.py \
+    -gtxt data/text \
+    -ptxt exp/cls_<split>/text \
+    -pscore exp/cls_<split>/score \
+    -tok data/token_list
+
+```
+
+## About data directory
+
+Each directory of training set, development set, and evaluation set, has same directory structure. See also https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE#about-kaldi-style-data-directory about Kaldi data structure.
+
+- Directory structure
+    ```
+    data/
+    ├── train/     # Training set directory
+    │   ├── text       # The transcription
+    │   ├── wav.scp    # Wave file path
+    │   ├── utt2spk    # A file mapping utterance-id to speaker-id
+    │   ├── spk2utt    # A file mapping speaker-id to utterance-id
+    |
+    ├── dev/
+    │   ...
+    ├── eval/
+    │   ...
+    └── token_list   # token list file
+        ...
+    ```
+
+ - `text` format
+    ```
+    uttidA <class_a>
+    uttidB <class_b1> <class_b2>
+    ...
+    ```
+    Note that for multi-class classification each uttid should be associated with exactly one class.
+    For multi-label classification, each uttid should have at least one label.
+    (TODO) We will support the case with no label in the future with the <blank> symbol.
+
+- `wav.scp` format
+    ```
+    uttidA /path/to/uttidA.wav
+    uttidB /path/to/uttidB.wav
+    ...
+    ```
+
+- `utt2spk` format
+    ```
+    uttidA speakerA
+    uttidB speakerB
+    uttidC speakerA
+    uttidD speakerB
+    ...
+    ```
+
+- `spk2utt` format
+    ```
+    speakerA uttidA uttidC ...
+    speakerB uttidB uttidD ...
+    ...
+    ```
+
+    Note that `spk2utt` file can be generated by `utt2spk`, and `utt2spk` can be generated by `spk2utt`, so it's enough to create either one of them.
+
+    ```bash
+    utils/utt2spk_to_spk2utt.pl data/train/utt2spk > data/train/spk2utt
+    utils/spk2utt_to_utt2spk.pl data/train/spk2utt > data/train/utt2spk
+    ```
+
+    If your corpus doesn't include speaker information, give the same speaker id as the utterance id to satisfy the directory format, otherwise give the same speaker id for all utterances (Actually we don't use speaker information for cls1 recipe now).
+
+    ```bash
+    uttidA uttidA
+    uttidB uttidB
+    ...
+    ```
+
+    OR
+
+    ```bash
+    uttidA dummy
+    uttidB dummy
+    ...
+    ```
+
+Once you complete creating the data directory, it's good to check it by `utils/validate_data_dir.sh`.
+
+```bash
+utils/validate_data_dir.sh --no-feats data/train
+utils/validate_data_dir.sh --no-feats data/dev
+utils/validate_data_dir.sh --no-feats data/test
+```
+
+### Problems you might encounter
+
+Below are some common errors to watch out for:
+
+#### 1. Torcheval not found
+* Run `pip install torcheval`
```

Comment:
Probably good to add in the installers and makefile?

Reply:
Done. Could you please check this once. I am not very familiar with makefiles, but this should be a simple one: https://github.com/espnet/espnet/pull/5998/commits/e6e9a160f13282f393ae397811df37d210bbbc92